homeassistant/components/buienradar/camera.py

```diff
@@ -0,0 +1,176 @@
+
+CONF_DIMENSION = 'dimension'
+CONF_DELTA = 'delta'
+
+RADAR_MAP_URL_TEMPLATE = ('https://api.buienradar.nl/image/1.0/'
+                          'RadarMapNL?w={w}&h={h}')
+
+_LOG = logging.getLogger(__name__)
+
+# Maximum range according to docs
+DIM_RANGE = vol.All(vol.Coerce(int), vol.Range(min=120, max=700))
+
+PLATFORM_SCHEMA = vol.All(
+    PLATFORM_SCHEMA.extend({
+        vol.Optional(CONF_DIMENSION): DIM_RANGE,
+        vol.Optional(CONF_DELTA): vol.All(vol.Coerce(float),
+                                          vol.Range(min=0)),
+        vol.Optional(CONF_NAME): cv.string,
+    }))
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up buienradar radar-loop camera component."""
+    c_id = config.get(CONF_ID)
+    dimension = config.get(CONF_DIMENSION) or 512
+    delta = config.get(CONF_DELTA) or 600.0
+    name = config.get(CONF_NAME) or "Buienradar loop"
+
+    async_add_entities([BuienradarCam(hass, name, c_id, dimension, delta)])
+
+
+class BuienradarCam(Camera):
+    """
+    A camera component producing animated buienradar radar-imagery GIFs.
+
+    Rain radar imagery camera based on image URL taken from [0].
+
+    [0]: https://www.buienradar.nl/overbuienradar/gratis-weerdata
+    """
+
+    _dimension = 0          # type: int
+    """ Deadline for image refresh """
+    _deadline = None        # type: Optional[datetime]
+    _name = ""              # type: str
+    _delta = 0.0         # type: float
+    _condition = None       # type: Optional[asyncio.Condition]
+    """ Loading status """
+    _loading = False        # type: bool
+
+    _last_image = None      # type: Optional[bytes]
+    """ last modified HTTP response header"""
+    _last_modified = None   # type: Optional[str]
+
+    def __init__(self, hass, name: str, c_id: Optional[str], dimension: int,
+                 delta: float):
+        """Initialize the component."""
+        super().__init__()
+        self._hass = hass
+        self._name = name
+
+        self._dimension = dimension
+        self._delta = delta
+
+        self._condition = asyncio.Condition(loop=hass.loop)
+
+        self._last_image = None
+        self._last_modified = None
+        self._loading = False
+        # deadline for image to be refreshed
+        self._deadline = None
+
+    @property
+    def name(self) -> str:
+        """Return the component name."""
+        return self._name
+
+    # Cargo-cult from components/proxy/camera.py
+    def camera_image(self) -> Optional[bytes]:
+        """Return camera image."""
+        return run_coroutine_threadsafe(
+            self.async_camera_image(), self.hass.loop).result()
+
+    def __needs_refresh(self) -> bool:
+        if not (self._delta and self._deadline and self._last_image):
+            return True
+
+        return dt_util.utcnow() > self._deadline
+
+    async def __retrieve_radar_image(self) -> bool:
```

Comment:
Convention is that for private methods, use a single underscore prefix. Double underscore prefix is reserved for Python built-in methods.

Reply:
I used double underscores since [name mangling](https://docs.python.org/3.7/tutorial/classes.html#private-variables) prevents conflicts with function names from parent classes.

This is different from `__[builtin]__`which has two trailing underscores. I will adjust this for `__retrieve_radar_image` but would prefer to keep it for the other case if mangled names are allowed in this project's style.

Reply:
Oh, TIL.